src/rust/receiver-mock/src/router.rs

```diff
@@ -167,6 +173,164 @@ pub async fn handler_terraform_fields_quota() -> impl Responder {
     })
 }
 
+pub async fn handler_terraform_fields(
+    terraform_state: web::Data<TerraformState>,
+) -> impl Responder {
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    struct TerraformFieldObject {
+        field_name: String,
+        field_id: String,
+        data_type: String,
+        state: String,
+    }
+
+    #[derive(Serialize)]
+    struct TerraformFieldsResponse {
+        data: Vec<TerraformFieldObject>,
+    }
+
+    let fields = terraform_state.fields.lock().unwrap();
+    let res = fields.iter().map(|(id, name)| TerraformFieldObject {
+        field_name: name.clone(),
+        field_id: id.clone(),
+        data_type: String::from("String"),
+        state: String::from("Enabled"),
+    });
+
+    web::Json(TerraformFieldsResponse {
+        data: res.collect(),
+    })
+}
+
+#[derive(Deserialize)]
+pub struct TerraformFieldParams {
+    field: String,
+}
+
+pub async fn handler_terraform_field(
+    params: web::Path<TerraformFieldParams>,
+    terraform_state: web::Data<TerraformState>,
+) -> impl Responder {
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    struct TerraformFieldResponse {
+        field_name: String,
+        field_id: String,
+        data_type: String,
+        state: String,
+    }
```

Comment:
Duplication of https://github.com/SumoLogic/sumologic-kubernetes-tools/pull/75/files#diff-f45289341f4631ac823eb8b9fdc542eb2200f8d3c4734c554442f19cb3c1e539R179-R186

Reply:
Please move it in common place

Reply:
done